pkg/observer/pvc_alt_test.go

```diff
@@ -0,0 +1,226 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observer
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/dell/cert-csi/pkg/k8sclient"
+	k8sclientV1 "github.com/dell/cert-csi/pkg/k8sclient/resources/pvc"
+	"github.com/dell/cert-csi/pkg/store"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+func TestPvcListObserver_StartWatching(t *testing.T) {
+	ctx := context.Background()
+
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-storage-class"},
+		VolumeBindingMode: func() *storagev1.VolumeBindingMode {
+			mode := storagev1.VolumeBindingWaitForFirstConsumer
+			return &mode
+		}(),
+	}
+
+	clientSet := NewFakeClientsetWithRestClient(storageClass)
+
+	clientSet.StorageV1().StorageClasses().Create(ctx, storageClass, metav1.CreateOptions{})
+
+	kubeClient := &k8sclient.KubeClient{
+		ClientSet: clientSet,
+		Config:    &rest.Config{},
+	}
+
+	pvcClient, _ := kubeClient.CreatePVCClient("test-namespace")
+
+	pvcsList := &v1.PersistentVolumeClaimList{
+		Items: []v1.PersistentVolumeClaim{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "test-pvc",
+					Namespace:         "test-namespace",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+				Status: v1.PersistentVolumeClaimStatus{
+					Phase: v1.ClaimBound,
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "test-pvc",
+					Namespace:         "test-namespace",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+				Status: v1.PersistentVolumeClaimStatus{
+					Phase: v1.ClaimBound,
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "test-pvc",
+					Namespace:         "test-namespace",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+				Status: v1.PersistentVolumeClaimStatus{
+					Phase: v1.ClaimBound,
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name    string
+		runner  *Runner
+		pvcList *v1.PersistentVolumeClaimList
+	}{
+		{
+			name: "Test case: Watching container pvc with driver namespace and metricList",
+			runner: &Runner{
+				Clients: &k8sclient.Clients{
+					PVCClient: pvcClient,
+				},
+				TestCase: &store.TestCase{
+					ID: 1,
+				},
+				WaitGroup:       sync.WaitGroup{},
+				DriverNamespace: "test-namespace",
+				Database:        NewSimpleStore(),
+			},
+			pvcList: pvcsList,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var pvcListWg sync.WaitGroup
+
+			ctx := context.Background()
+
+			test.runner.WaitGroup.Add(1)
+
+			po := &PvcListObserver{}
+			po.MakeChannel()
+
+			if test.pvcList != nil {
+				pvcListWg.Add(1)
+				originalGetPvcsList := getPvcsList
```

Comment:
Good practice to name getPvcList as something like getPVCListFunc, helps for better readability of the code.

Reply:
done!